### PR TITLE
1007 refactor classes to use commons collection 4 as collections 3 is flagged as vulnerable new

### LINKED ISSUE #1007: Refactor classes to use commons collection 4 as collections 3 is flagged as vulnerable
Refactor the following classes to use commons collections 4.  Update dependency reference in pom to use commons collection4 instead of commons collections. 

- modules/utils/src/test/java/com/percussion/util/FunctionalUtilsTest.java
- modules/utils/src/test/java/com/percussion/utils/PSItemIteratorTest.java
- modules/utils/src/main/java/com/percussion/utils/collections/PSMultiMapIterator.java
- modules/utils/src/test/java/com/percussion/utils/collections/PSMultiMapIterTest.java
- modules/utils/src/main/java/com/percussion/utils/jsr170/PSItemIterator.java
- modules/utils/src/main/java/com/percussion/utils/jsr170/PSNamePatternFilter.java
- modules/utils/src/main/java/com/percussion/utils/jsr170/PSNodeIterator.java

### PR DESCRIPTION
**Pull request by rgarg24**

### CODE REVIEW

File: modules/utils/pom.xml [hunk 1]
```diff
@@ -171,6 +171,12 @@
             <artifactId>commons-collections</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-collections4</artifactId>
+            <version>${commons.collections4.version}</version>
```
Comment (natechadwick) [inline modules/utils/pom.xml]: You shouldn't need the version here as it is defined in the parent pom.xml

File: modules/utils/pom.xml [hunk 2]
```diff
@@ -171,6 +171,12 @@
             <artifactId>commons-collections</artifactId>
```
Comment (natechadwick) [inline modules/utils/pom.xml]: Can we remove the commons collections 3 dependency?  The security warnings won't get dropped for that dependency unless it is removed.

File: modules/utils/pom.xml (hunk 1 above)
Comment (rgarg24) [reply modules/utils/pom.xml]: Yes Nate done the required changes and pushed the code.

File: modules/utils/pom.xml (hunk 2 above)
Comment (rgarg24) [reply modules/utils/pom.xml]: Yes Nate removed the commons collections 3 dependency and pushed the code.

**Review by natechadwick — APPROVED** [review]

**Pull request merged**